contracts/peripheral/AuraClaimZapV2.sol

```diff
@@ -0,0 +1,280 @@
+
+        //claim from main curve LP pools
+        for (uint256 i = 0; i < rewardContracts.length; i++) {
+            IRewardStaking(rewardContracts[i]).getReward(msg.sender, true);
+        }
+        //claim from extra rewards
+        for (uint256 i = 0; i < extraRewardContracts.length; i++) {
+            IRewardStaking(extraRewardContracts[i]).getReward(msg.sender);
+        }
+        //claim from multi reward token contract
+        for (uint256 i = 0; i < tokenRewardContracts.length; i++) {
+            IRewardStaking(tokenRewardContracts[i]).getReward(msg.sender, tokenRewardTokens[i]);
+        }
+
+        //claim from cvxCrv rewards
+        if (options.claimCvxCrv) {
+            IRewardStaking(cvxCrvRewards).getReward(msg.sender, true);
+        }
+
+        //claim from locker
+        if (options.claimLockedCvx) {
+            IAuraLocker(locker).getReward(msg.sender);
+        }
+
+        // deposit/lock/stake
+        if (_callRelockRewards(options)) {
+            _relockRewards(crvBalance, cvxBalance, cvxCrvBalance, amounts, options);
+        }
+    }
+
+    /**
+     * @notice returns a bool if relocking of rewards should occur
+     * @param options                Claim options
+     */
+    function _callRelockRewards(Options calldata options) internal view returns (bool) {
+        return (options.lockCvxCrv || options.lockCrvDeposit || options.useCompounder || options.lockCvx);
+    }
+
+    /**
+     * @notice  Claim additional rewards from:
+     *          - cvxCrvRewards
+     *          - cvxLocker
+     * @param removeCrvBalance       crvBalance to ignore and not redeposit (starting Crv balance)
+     * @param removeCvxBalance       cvxBalance to ignore and not redeposit (starting Cvx balance)
+     * @param removeCvxCrvBalance    cvxcrvBalance to ignore and not redeposit (starting CvxCrv balance)
+     * @param amounts                Claim rewards amoutns.
+     * @param options                see claimRewards
+     */
+    // prettier-ignore
+    function _relockRewards( // solhint-disable-line 
+        uint256 removeCrvBalance,
+        uint256 removeCvxBalance,
+        uint256 removeCvxCrvBalance,          
+        ClaimRewardsAmounts calldata amounts, 
+        Options calldata options
+    ) internal {
+        
+        //lock upto given amount of crv as cvxCrv
+        if (amounts.depositCrvMaxAmount > 0) {
+            (uint256 crvBalance, bool continued) = _checkBalanceAndPullToken(crv, removeCrvBalance, amounts.depositCrvMaxAmount);
+
+            if (continued) {ICrvDepositorWrapper(crvDepositWrapper).deposit(
+                    crvBalance,
+                    amounts.minAmountOut,
+                    options.lockCrvDeposit,
+                    address(0)
+                );}
+        }
+
+        
+        //Pull cvxCrv to contract if user wants to stake
+        if (options.lockCvxCrv) {
+            _checkBalanceAndPullToken(cvxCrv, removeCvxCrvBalance, amounts.depositCvxCrvMaxAmount);
```

Comment:
Why here it is not use the result of the function ? ```(uint256 crvBalance, bool continued)```

Reply:
For this option, we only need to calculate the no. tokens to pull in and then pull them in. The deposit step occurs at L243 in order to avoid 2 deposits/stakes happening in 1 txn.